### ISSUE TITLE
API for allowing plugins to access other plugins

### ISSUE THREAD

**Issue by jthomas** [serverless/serverless#2785]
Working on the OpenWhisk provider implementation for Serverless I've hit a challenge trying to access other plugin instances from within my plugins. 

In the *deployFunction* plugin, we need to access the existing *package* plugin to manage packaging a single function artifact.

In the AWS provider implementation, the class can directly reference the module through a local file path. https://github.com/serverless/serverless/blob/master/lib/plugins/aws/deployFunction/index.js#L9

The OpenWhisk plugin is a third-party plugin and cannot do this. 

I'm currently working around this by iterating over the installed plugin instances, from the plugin manager, but this relies on monkey-testing the instance to assume we have the correct instance. 
https://github.com/serverless/serverless-ibm-openwhisk/blob/master/deployFunction/index.js#L15-L16

Could we add some API method to help me find installed plugins by their name?

**Comment by pmuens** [comment]
@jthomas thank you for opening this! 👍  Perfect timing 🕐 

We're currently discussing how we can improve the overall plugin architecture / UX.
Great to have such a feedback about current pain points (please submit more if you have anything else). 💯 

/cc @ac360 @nikgraf @eahefnawy @mthenw

**Comment by nikgraf** [comment]
Indeed, very good feedback! We make sure this is possible in the future.

**Comment by medikoo** [comment]
I've updated main description with implementation proposal. It can be achieved quite simply (I think), and added value is real
